### PR TITLE
fix(appstore-connect): Fix empty dSYMUrl detection

### PR DESCRIPTION
**Pull request by relaxolotl**
https://github.com/getsentry/sentry/pull/29704's check for the presence of dSYMS was slightly off, see SENTRY-SN6. The code uses an enum to represent cases where there is no dSYM present, so the truthy check for a url would always pass, causing app clip dsyms to always be reported.

This adds tests to verify the logic as well.

### CODE REVIEW

File: tests/sentry/utils/appleconnect/test_appstore_connect.py [hunk 1]
```diff
@@ -6,17 +8,18 @@
 
 class TestGetDsymUrl:
     def test_none_bundles(self) -> None:
-        assert appstore_connect._get_dsym_url(None) is NoDsymUrl.NOT_NEEDED
+        assert appstore_connect._get_dsym_url(None) == NoDsymUrl.NOT_NEEDED
```
Comment (flub) [inline tests/sentry/utils/appleconnect/test_appstore_connect.py:11]: these comparisons were correct with `is` afaik, why did you change them?
https://docs.python.org/3/library/enum.html?#comparisons

**Review by flub — APPROVED** [review]

**Review by loewenheim — APPROVED** [review]
